src/components/ftbot/TradeList.vue

```diff
@@ -44,41 +46,85 @@ const removeTradeVisible = ref(false);
 const confirmExitText = ref('');
 const confirmExitValue = ref<ModalReasons | null>(null);
 
+// NEW: Menu reference and items for column toggle dropdown
+const menu = ref(null);
+
+
+
+
+// Modified: Add visible property to tableFields and initialize menuItems
+
 const increasePosition = ref({ visible: false, trade: {} as Trade });
 function formatPriceWithDecimals(price: number) {
   return formatPrice(price, botStore.activeBot.stakeCurrencyDecimals);
 }
 
+// Modified: Define tableFields with visible property
 const tableFields = ref([
-  { field: 'trade_id', header: 'ID' },
-  { field: 'pair', header: 'Pair' },
-  { field: 'amount', header: 'Amount' },
+  { field: 'trade_id', header: 'ID', visible: true },
+  { field: 'pair', header: 'Pair', visible: true },
+  { field: 'amount', header: 'Amount', visible: true },
   props.activeTrades
-    ? { field: 'stake_amount', header: 'Stake amount' }
-    : { field: 'max_stake_amount', header: 'Total stake amount' },
+    ? { field: 'stake_amount', header: 'Stake amount', visible: true }
+    : { field: 'max_stake_amount', header: 'Total stake amount', visible: true },
   {
     field: 'open_rate',
     header: 'Open rate',
+    visible: true,
   },
   {
     field: props.activeTrades ? 'current_rate' : 'close_rate',
     header: props.activeTrades ? 'Current rate' : 'Close rate',
+    visible: true,
   },
   {
     field: 'profit',
     header: props.activeTrades ? 'Current profit %' : 'Profit %',
+    visible: true,
   },
-  { field: 'open_timestamp', header: 'Open date' },
+  { field: 'open_timestamp', header: 'Open date', visible: true },
   ...(props.activeTrades
-    ? [{ field: 'actions', header: '' }]
+    ? [{ field: 'actions', header: '', visible: true }]
     : [
-        { field: 'close_timestamp', header: 'Close date' },
-        { field: 'exit_reason', header: 'Close Reason' },
+        { field: 'close_timestamp', header: 'Close date', visible: true },
+        { field: 'exit_reason', header: 'Close Reason', visible: true },
       ]),
 ]);
 
+const menuItems = ref([
+  {
+    label: 'Toggle Columns',
+    items: tableFields.value.map(col => ({
+      label: col.header,
+      command: () => toggleColumn(col.field),
+      item: ({ item }) => h(
+        'div',
+        { class: 'p-menuitem-content flex align-items-center' },
+        [
+          h(BaseCheckbox, {
+            modelValue: col.visible,
+            'onUpdate:modelValue': () => toggleColumn(col.field),
+            class: 'p-checkbox-input',
```

Comment:
using p- classes is no longer state of the art for primevue4.
while it still works, it shouldn't be used. Obviously, the same also applies to the bottom part.